### ISSUE TITLE
Build mini version 2B with smd components and Atmega328p chip. But all sd cards are No dos recognised!

### ISSUE THREAD

**Issue by Alectronics** [jbilander/SDBox-v2#7]
Only the card change is detected but never any transfer or data led activity visible. Only my green power led is always lit.
Tried every driver/boot version I could think of on my Amiga 1200 but the SD reader interface is very unreliable and never works as was expected. I also tried a 2GB fat and fat32 formatted original Sandisk microSD card same issue. No-dos disk.
Boards were made by PCBway according to the v2b gerber files but all was a waste of time and money. 
Many other buyers/makers made this same comment. And the solutions given are always useless. After they gave up on this product!
This interface in case of these hopeless issues should have many more presented diagnostic info to check on like scope diagrams or logic analyzer diagrams to fix a not working interface.

Luckily my CF adapter interface always works splendidly to transfer files between my PC and Amiga. Including a perfect working internal CF Windows boot 'hdd'.

**Comment by jbilander** [comment]
Hi, I have built several v2 and Mini devices with PCBs from JLCPCB and they all work perfectly fine with the latest fw and driver:

https://github.com/jbilander/amiga-par-to-spi-adapter/

There is no problem with the hardware. It has been tested by many people by now with the above firmware and it works even on a wirewrap thingie.

There has been reports of some MicroSD-cards not working properly and problems on the A1000.
https://github.com/niklasekstrom/amiga-par-to-spi-adapter/issues/8

And of course the usual thing with checking the CIAs with ATK and put pull-ups on the parallel-port control- and data-lines (required for some A3000 it seems)

_"This interface in case of these hopeless issues should have many more presented diagnostic info to check on like scope diagrams or logic analyzer diagrams to fix a not working interface."_

So why do you not make them? Remember this is an open source hobby project. Everything is there for anyone to dig in and help to improve it. Contribute nothing, expect nothing!

**Comment by Alectronics** [comment]
Yes I understand but sorry Jorgen. I already wasted over 2 weeks figuring out why it won't work invain.

I even questioned the Amiga 1200 parallel port being defect why I also build today a parallel loop port tester and found out that it was working 100% fine!

So nothing is defect only the still not working v2B mini sd card reader/writer must be defect or badly designed.
Sadly I can't fix a problem that is made by others who obviously spended a lot of time to try to build such genious device that most of the time gives compatibility issues if it works!

Albert.

________________________________
Van: Jorgen Bilander ***@***.***>
Verzonden: zaterdag 5 augustus 2023 11:53
Aan: jbilander/SDBox-v2 ***@***.***>
CC: Alectronics ***@***.***>; Author ***@***.***>
Onderwerp: Re: [jbilander/SDBox-v2] Build mini version 2B with smd components and Atmega328p chip. But all sd cards are No dos recognised! (Issue #7)


Hi, I have built several v2 and Mini devices with PCBs from JLCPCB and they all work perfectly fine with the latest fw and driver:

https://github.com/jbilander/amiga-par-to-spi-adapter/

There is no problem with the hardware. It has been tested by many people by now with the above firmware and it works even on a wirewrap thingie.

There has been reports of some MicroSD-cards not working properly and problems on the A1000.
niklasekstrom/amiga-par-to-spi-adapter#8<https://github.com/niklasekstrom/amiga-par-to-spi-adapter/issues/8>

And of course the usual thing with checking the CIAs with ATK and put pull-ups on the parallel-port control- and data-lines (required for some A3000 it seems)

"This interface in case of these hopeless issues should have many more presented diagnostic info to check on like scope diagrams or logic analyzer diagrams to fix a not working interface."

So why do you not make them? Remember this is an open source hobby project. Everything is there for anyone to dig in and help to improve it. Contribute nothing, expect nothing!

**Comment by jbilander** [comment]
So you need to do basic troubleshooting:

* Does the CIA work correctly with ATK (Amiga Test Kit)
* Is 5V-powering to the device looking good?
* Is 3.3V generated correctly to the MicroSD-card holder VCC-pin?
* Do you have activity on the MISO-line?
* Do you have activity on the MOSI-line?
* Do you have activity on the Parallel-port data and control lines? Pull-ups might be needed.
* Is the latest firmware flashed to the ATMega328p with fuses correctly set?
* Is the latest driver spisd.device put in the Devs folder on the Amiga?
* Is the correct version of the SD0 Mount file used for your WB-version on the Amiga?
* Is the latest fat95 driver copied to the Amiga?
* Does the 16 MHz clock looks ok going to the ATMega input pin?
* Do you have any solder bridges?
* Are you using a known good MicroSD-card?
* Does the device show the same error on another Amiga?
* Is the MicroSD-card properly FAT/FAT32-formatted?
etc, etc...

**Comment by Alectronics** [comment]
Thanks for your fast reply Jorgen. But I thought my SDbox 2.b interface was complete and didn't need any pullups or modification?
What more do I also need to pullup because those smd 4R10K pullups were already on the miniSD reader board to accomplish that?

Albert.



________________________________
Van: Jorgen Bilander ***@***.***>
Verzonden: zaterdag 5 augustus 2023 12:34
Aan: jbilander/SDBox-v2 ***@***.***>
CC: Alectronics ***@***.***>; Author ***@***.***>
Onderwerp: Re: [jbilander/SDBox-v2] Build mini version 2B with smd components and Atmega328p chip. But all sd cards are No dos recognised! (Issue #7)


So you need to do basic troubleshooting:

  *   Does the CIA work correctly with ATK (Amiga Test Kit)
  *   Is 5V-powering to the device looking good?
  *   Is 3.3V generated correctly to the MicroSD-card holder VCC-pin?
  *   Do you have activity on the MISO-line?
  *   Do you have activity on the MOSI-line?
  *   Do you have activity on the Parallel-port data and control lines? Pull-ups might be needed.
  *   Is the latest firmware flashed to the ATMega328p with fuses correctly set?
  *   Is the latest driver spisd.device put in the Devs folder on the Amiga?
  *   Is the correct version of the SD0 Mount file used for your WB-version on the Amiga?
  *   Is the latest fat95 driver copied to the Amiga?
  *   Does the 16 MHz clock looks ok going to the ATMega input pin?
  *   Do you have any solder bridges?
  *   Are you using a known good MicroSD-card?
  *   Does the device show the same error on another Amiga?
  *   Is the MicroSD-card properly FAT/FAT32-formatted?
etc, etc...

**Comment by jbilander** [comment]
pull-ups should not be needed on the A1200. I think it was required for the A3000.

https://github.com/jbilander/SDBox-v2/issues/4#issuecomment-1359937696

**Comment by Alectronics** [comment]
Understood. I have tried all software drivers and boot disks I could find. My Amiga 1200 also has an expansion board from manufacturer 8738 product 35. Kickstart v39.106, WB v40.42. 2Meg AGA 9.21 MFlops. But even when booted from DF0: without using the internal CF boot WB hdd it made no difference.
My Amiga only once with a lot of read errors recognized the 2GB Sandisk but I could not use it. Now that micro SD card never works anymore! And I also tried other SD cards with a SD card to microSD card adapter but on those not even the card change worked!

And I am a very experience electronics devices maker especially when it is about soldering. So that never will be it! It maybe is the tiniest 16MHz 4 'pins' smd quartz tht I used that better should be a lager quartz? But it has worked before and those larger 5070 smd size crystals are ordered but not here yet.

There are too much uncertainties involved in the now different versions available which makes finding the problem even harder! Why I probably give up on this project, and keep using my Superb working multicard microSD to CF adapter for file transfering from PC to Amiga and vice-versa. That works with all Micro-SD card no matter what brand!

Best regards.
Albert.


________________________________
Van: Jorgen Bilander ***@***.***>
Verzonden: zaterdag 5 augustus 2023 12:49
Aan: jbilander/SDBox-v2 ***@***.***>
CC: Alectronics ***@***.***>; Author ***@***.***>
Onderwerp: Re: [jbilander/SDBox-v2] Build mini version 2B with smd components and Atmega328p chip. But all sd cards are No dos recognised! (Issue #7)


pull-ups should not be needed on the A1200. I think it was required for the A3000.

#4 (comment)<https://github.com/jbilander/SDBox-v2/issues/4#issuecomment-1359937696>

**Comment by jbilander** [comment]
why? Just build the latest firmware and driver from source and use that. That is the one you want with V2/Mini, not any other older version or combination of versions.

If your MicroSD-card is fried you probably have exposed it to too high voltage by accident, above 3.3V. Perhaps you have a short somewhere but if you don't troubleshoot you will never find it. I have heard PCBWay sometimes even send out boards with shorts in them but I haven't used them myself so no personal experience with that. I always check the PCB with a multimeter before I even start to populate it with components.

If you already have a superior way of transferring files why even bother with the SDBox. Just continue using the superior way and stop wasting other peoples time complaining. It's a DIY-hobby project that is made for fun by tinkerers for tinkerers and the Amiga community. Bad attitude will not exactly put you first in the support line if you know what I mean. It's easy to complain and demand things from others but actually doing constructive work and contribute something yourself is way harder, isn't it? You know amigastore even sell SDBox-v2 units so the hardware/fw cannot be that bad, can it?, or are all the devices they sell not working? BTW, I have nothing to do with them selling units, nor do Niklas.

**Comment by Alectronics** [comment]
I know Jorgen. I used the newest drivers with the multilanguage FAT95 files on them.
After which my SD0: CF drive was gone and disabled.

So more trouble here sadly. Why I placed the old FAT95 back and removed that Mini micro SD Box driver  version. Its useless to try to fix something so unreliable.

Have a good weekend!
Albert.
________________________________
Van: Jorgen Bilander ***@***.***>
Verzonden: zaterdag 5 augustus 2023 14:11
Aan: jbilander/SDBox-v2 ***@***.***>
CC: Alectronics ***@***.***>; Author ***@***.***>
Onderwerp: Re: [jbilander/SDBox-v2] Build mini version 2B with smd components and Atmega328p chip. But all sd cards are No dos recognised! (Issue #7)


why? Just build the latest firmware and driver from source and use that. That is the one you want with V2/Mini, not any other older version or combination of versions.

If your MicroSD-card is fried you probably have exposed it to too high voltage by accident, above 3.3V. Perhaps you have a short somewhere but if you don't troubleshoot you will never find it. I have heard PCBWay sometimes even send out boards with shorts in them but I haven't used them myself so no personal experience with that. I always check the PCB with a multimeter before I even start to populate it with components.

If you already have a superior way of transferring files why even bother with the SDBox. Just continue using the superior way and stop wasting other peoples time complaining. It's a DIY-hobby project that is made for fun by tinkerers for tinkerers and the Amiga community. Bad attitude will not exactly put you first in the support line if you know what I mean. It's easy to complain and demand things from others but actually doing constructive work and contribute something yourself is way harder, isn't it? You know amigastore even sell SDBox-v2 units so the hardware/fw cannot be that bad, can it?, or are all the devices they sell not working? BTW, I have nothing to do with them selling units, nor do Niklas.

**Comment by Alectronics** [comment]
By-the-way even with the new SDbox drivers from here: https://github.com/jbilander/SDBox-v2/releases/tag/v2a my Amiga 1200 never shows the SDcard disk icon anymore either. Only the sd card switch detect works. SD cards even the Sandisk 2GB FAT32 ones are always NoDos and invisible now! Although my Amiga rapports that the SD card driver is already mounted. Cd SD0: only gets NoDOS problem SD cards!
[https://opengraph.githubassets.com/83c45951c5e62f758c190aa5170acd920fe282420b2104e292fda476be27058c/jbilander/SDBox-v2/releases/tag/v2a]<https://github.com/jbilander/SDBox-v2/releases/tag/v2a>
Release Rev. 2A · jbilander/SDBox-v2<https://github.com/jbilander/SDBox-v2/releases/tag/v2a>
Tag for Revision 2A.
github.com

________________________________
Van: Albert van bemmelen ***@***.***>
Verzonden: zaterdag 5 augustus 2023 14:21
Aan: jbilander/SDBox-v2 ***@***.***>
Onderwerp: Re: [jbilander/SDBox-v2] Build mini version 2B with smd components and Atmega328p chip. But all sd cards are No dos recognised! (Issue #7)

I know Jorgen. I used the newest drivers with the multilanguage FAT95 files on them.
After which my SD0: CF drive was gone and disabled.

So more trouble here sadly. Why I placed the old FAT95 back and removed that Mini micro SD Box driver  version. Its useless to try to fix something so unreliable.

Have a good weekend!
Albert.
________________________________
Van: Jorgen Bilander ***@***.***>
Verzonden: zaterdag 5 augustus 2023 14:11
Aan: jbilander/SDBox-v2 ***@***.***>
CC: Alectronics ***@***.***>; Author ***@***.***>
Onderwerp: Re: [jbilander/SDBox-v2] Build mini version 2B with smd components and Atmega328p chip. But all sd cards are No dos recognised! (Issue #7)


why? Just build the latest firmware and driver from source and use that. That is the one you want with V2/Mini, not any other older version or combination of versions.

If your MicroSD-card is fried you probably have exposed it to too high voltage by accident, above 3.3V. Perhaps you have a short somewhere but if you don't troubleshoot you will never find it. I have heard PCBWay sometimes even send out boards with shorts in them but I haven't used them myself so no personal experience with that. I always check the PCB with a multimeter before I even start to populate it with components.

If you already have a superior way of transferring files why even bother with the SDBox. Just continue using the superior way and stop wasting other peoples time complaining. It's a DIY-hobby project that is made for fun by tinkerers for tinkerers and the Amiga community. Bad attitude will not exactly put you first in the support line if you know what I mean. It's easy to complain and demand things from others but actually doing constructive work and contribute something yourself is way harder, isn't it? You know amigastore even sell SDBox-v2 units so the hardware/fw cannot be that bad, can it?, or are all the devices they sell not working? BTW, I have nothing to do with them selling units, nor do Niklas.

**Comment by jbilander** [comment]
Did I not say build the latest firmware and driver from source and use that. If you don't wanna listen sorry I can't help you.
There have been bugfixes to the driver since that release.
You can read about it here:
https://github.com/niklasekstrom/amiga-par-to-spi-adapter/commits?author=patrikaxelsson

**Comment by Alectronics** [comment]
Sorry Jorgen.
I never understood that building the firmware thing with the VBCC software.
I tried to use it but had too little information to go on.

Albert.
________________________________
Van: Jorgen Bilander ***@***.***>
Verzonden: zaterdag 5 augustus 2023 16:07
Aan: jbilander/SDBox-v2 ***@***.***>
CC: Alectronics ***@***.***>; Author ***@***.***>
Onderwerp: Re: [jbilander/SDBox-v2] Build mini version 2B with smd components and Atmega328p chip. But all sd cards are No dos recognised! (Issue #7)


Did I not say build the latest firmware and driver from source and use that. If you don't wanna listen sorry I can't help you.
There have been bugfixes to the driver since that release.
You can read about it here:
https://github.com/niklasekstrom/amiga-par-to-spi-adapter/commits?author=patrikaxelsson

**Comment by Alectronics** [comment]
PS: FYI latest version 2.2B presented here also tested not working!:
https://github.com/niklasekstrom/amiga-par-to-spi-adapter/releases/tag/v2.2
[https://opengraph.githubassets.com/c614ff12f02d161ae280684b09908629ba9b7484661a3edc8c5755c6382f9544/niklasekstrom/amiga-par-to-spi-adapter/releases/tag/v2.2]<https://github.com/niklasekstrom/amiga-par-to-spi-adapter/releases/tag/v2.2>
Release Release v2.2 with support for large SD cards · niklasekstrom/amiga-par-to-spi-adapter<https://github.com/niklasekstrom/amiga-par-to-spi-adapter/releases/tag/v2.2>
The SD card driver has been updated with support for large SD cards (> 4GB). What's Changed Large SD card support by @patrikaxelsson in #6 New Contributors @patrikaxelsson made their first contr...
github.com

________________________________
Van: Jorgen Bilander ***@***.***>
Verzonden: zaterdag 5 augustus 2023 16:07
Aan: jbilander/SDBox-v2 ***@***.***>
CC: Alectronics ***@***.***>; Author ***@***.***>
Onderwerp: Re: [jbilander/SDBox-v2] Build mini version 2B with smd components and Atmega328p chip. But all sd cards are No dos recognised! (Issue #7)


Did I not say build the latest firmware and driver from source and use that. If you don't wanna listen sorry I can't help you.
There have been bugfixes to the driver since that release.
You can read about it here:
https://github.com/niklasekstrom/amiga-par-to-spi-adapter/commits?author=patrikaxelsson

**Comment by jbilander** [comment]
it doesn't work for anyone...these videos are all fake?

https://youtu.be/mpZu9q7MV5I?t=754

https://drive.google.com/file/d/1WVmxK-hrYP19Keb9Lf7x2Z9d_c_A9ELw/view

**Comment by Alectronics** [comment]
Nice videos Jorgen, why I decided to make this too, but sadly without any success.

Maybe I got bad or fake ATmega328p China microcontrollers and/or bad microSD cards because it just never worked! Now it is not even showing the mini SDBox icon on the WB screen anymore. Only the microSD switch detect works on all SD cards. And never did the red activity led ever light up! And it is correctly connected to the ATmega pin with its cathode with the resistor in serie. Its anode to Vcc.
I also checked my PCB board and its copper tracks and all component connections from PCBway and couldn't find anything wrong whatsover!

If there in future is another firmware upgrade that also accepts all brand micro SD cards I will try it again. But for now I put this little project aside, including the other 4 still empty boards from PCBway China.
This was simply no fun.
Maybe I should also pullup all parallelport lines on my board like for A3000 Amiga users if that helps. But I'm not sure what pullups they already successfully did on the A3000.

Thanks,
Albert.


________________________________
Van: Jorgen Bilander ***@***.***>
Verzonden: zaterdag 5 augustus 2023 22:38
Aan: jbilander/SDBox-v2 ***@***.***>
CC: Alectronics ***@***.***>; Author ***@***.***>
Onderwerp: Re: [jbilander/SDBox-v2] Build mini version 2B with smd components and Atmega328p chip. But all sd cards are No dos recognised! (Issue #7)


it doesn't work for anyone...these videos are all fake?

https://youtu.be/mpZu9q7MV5I?t=754

https://drive.google.com/file/d/1WVmxK-hrYP19Keb9Lf7x2Z9d_c_A9ELw/view

**Comment by jbilander** [comment]
Well, if you have only tried ONE type of MiroSD-card and you say that card now does not even work in your PC or Mac. Maybe you should start thinking that it would be smart to try another card? Perhaps one listed here since they are verified to work. Makes me think you are not troubleshooting anything at all, just trolling.

https://github.com/jbilander/sdbox/issues/15#issuecomment-1120407423

The activity LED should work if you build from sources here:
https://github.com/jbilander/amiga-par-to-spi-adapter

specifically this is the change:
https://github.com/niklasekstrom/amiga-par-to-spi-adapter/commit/aeaf351bd57fc0b75dae18e1f8b91ef14cbc8c00

Niklas' code/release does not include code driving the activity LED. Anyway you should see activity on the MISO and MOSI lines and of course the parallel lines during MOUNT if you troubleshoot with a logic analyser or oscilloscope. I have already given you a list of what to start checking.

**Comment by Alectronics** [comment]
Sorry I already tried 6 different cards of which only one was microSD Sandisk 2GB. And the latter shortly worked with a lot of errors but now does nothing anymore.
My windows computers have no problem with reading any of them! Fat32 formatted. Only the miniSDbox v2b refuses to see them!


And sorry can't do anything with this info, since there is no driver FW =>

https://github.com/jbilander/amiga-par-to-spi-adapter

And I probably already tried all the versions 2 that had.

I'm pretty sure that it must be some software problem with the libraries or drivers. If it is not about port line pullups.

Albert.

________________________________
Van: Jorgen Bilander ***@***.***>
Verzonden: zondag 6 augustus 2023 00:04
Aan: jbilander/SDBox-v2 ***@***.***>
CC: Alectronics ***@***.***>; Author ***@***.***>
Onderwerp: Re: [jbilander/SDBox-v2] Build mini version 2B with smd components and Atmega328p chip. But all sd cards are No dos recognised! (Issue #7)


Well, if you have only tried ONE type of MiroSD-card and you say that card now does not even work in your PC or Mac. Maybe you should start thinking that it would be smart to try another card? Perhaps one listed here since they are verified to work. Makes me think you are not troubleshooting anything at all, just trolling.

jbilander/sdbox#15 (comment)<https://github.com/jbilander/sdbox/issues/15#issuecomment-1120407423>

The activity LED should work if you build from sources here:



Sorry can't do anything with this onfo, there is no driver FW =>

https://github.com/jbilander/amiga-par-to-spi-adapter

specifically this is the change:
***@***.***<https://github.com/niklasekstrom/amiga-par-to-spi-adapter/commit/aeaf351bd57fc0b75dae18e1f8b91ef14cbc8c00>

Niklas' code/release does not include code driving the activity LED. Anyway you should see activity on the MISO and MOSI lines and of course the parallel lines during MOUNT if you troubleshoot with a logic analyser or oscilloscope. I have already given you a list of what to start checking.

**Comment by Alectronics** [comment]
I've thrown this crap product aside after the miniSDbox continuesly failed to work. And after all this time of wasted testing also the micro-usb connector broke of the tiny copper layers for the third time. Which probably wouldn't have happened if it finally was working and safely ready build in the DB25 connector case.

Those that got lucky with building this Amiga miniSDBox reader must have been extremely fortunate or they could find the right firmware for it because I never had so much trouble with finding the right firmware or getting a device working before!
And those people in the happy videos obviously got it ready tested including the properly working library files. Because to me and other buyers who also claimed their ready bought Amiga parallel port reader never worked it looks like this unfinished project was designed to fail.





________________________________
Van: Albert van bemmelen ***@***.***>
Verzonden: zondag 6 augustus 2023 00:16
Aan: jbilander/SDBox-v2 ***@***.***>
Onderwerp: Re: [jbilander/SDBox-v2] Build mini version 2B with smd components and Atmega328p chip. But all sd cards are No dos recognised! (Issue #7)

Sorry I already tried 6 different cards of which only one was microSD Sandisk 2GB. And the latter shortly worked with a lot of errors but now does nothing anymore.
My windows computers have no problem with reading any of them! Fat32 formatted. Only the miniSDbox v2b refuses to see them!


And sorry can't do anything with this info, since there is no driver FW =>

https://github.com/jbilander/amiga-par-to-spi-adapter

And I probably already tried all the versions 2 that had.

I'm pretty sure that it must be some software problem with the libraries or drivers. If it is not about port line pullups.

Albert.

________________________________
Van: Jorgen Bilander ***@***.***>
Verzonden: zondag 6 augustus 2023 00:04
Aan: jbilander/SDBox-v2 ***@***.***>
CC: Alectronics ***@***.***>; Author ***@***.***>
Onderwerp: Re: [jbilander/SDBox-v2] Build mini version 2B with smd components and Atmega328p chip. But all sd cards are No dos recognised! (Issue #7)


Well, if you have only tried ONE type of MiroSD-card and you say that card now does not even work in your PC or Mac. Maybe you should start thinking that it would be smart to try another card? Perhaps one listed here since they are verified to work. Makes me think you are not troubleshooting anything at all, just trolling.

jbilander/sdbox#15 (comment)<https://github.com/jbilander/sdbox/issues/15#issuecomment-1120407423>

The activity LED should work if you build from sources here:



Sorry can't do anything with this onfo, there is no driver FW =>

https://github.com/jbilander/amiga-par-to-spi-adapter

specifically this is the change:
***@***.***<https://github.com/niklasekstrom/amiga-par-to-spi-adapter/commit/aeaf351bd57fc0b75dae18e1f8b91ef14cbc8c00>

Niklas' code/release does not include code driving the activity LED. Anyway you should see activity on the MISO and MOSI lines and of course the parallel lines during MOUNT if you troubleshoot with a logic analyser or oscilloscope. I have already given you a list of what to start checking.

**Comment by jbilander** [comment]
Not all people are made for DIY. Perhaps in your next life. Who knows, you might even be talented enough to run a bat file to compile a driver. Let's hope for that.

**Comment by Alectronics** [comment]
Not all people present the ready and working upgrade firmware either LOL. Because all seen Github pages failed presenting those files. And the many firmware changes you previously showed made clear that only the insiders know what it is all about.
If I only knew  WHAT bat file is the correct REALLY working version? Because all previous boot disks and version 2 files completely failed working!
________________________________
Van: Jorgen Bilander ***@***.***>
Verzonden: zondag 6 augustus 2023 12:03
Aan: jbilander/SDBox-v2 ***@***.***>
CC: Alectronics ***@***.***>; Author ***@***.***>
Onderwerp: Re: [jbilander/SDBox-v2] Build mini version 2B with smd components and Atmega328p chip. But all sd cards are No dos recognised! (Issue #7)


Not all people are made for DIY. Perhaps in your next life. Who knows, you might even be talented enough to run a bat file to compile a driver. Let's hope for that.

**Comment by jbilander** [comment]
All instructions are there, nothing is hidden, did you even look??? did you even try??? did you even check activity on the signals I told you to check??? Sorry to say but a DIY-project is not a thing for you if you need to be spoon-fed.

<img width="957" alt="Screen Shot 2023-08-06 at 12 37 00" src="https://github.com/jbilander/SDBox-v2/assets/1673918/e389c68f-04e8-42fb-a54b-86b1ca1df6e4">



https://github.com/jbilander/SDBox-v2/blob/main/images/SDBox-v2_VBCC_pic1.jpg

https://github.com/jbilander/SDBox-v2/blob/main/images/SDBox-v2_VBCC_pic2.jpg

https://github.com/jbilander/SDBox-v2/blob/main/images/SDBox-v2_VBCC_pic3.jpg

https://github.com/jbilander/SDBox-v2/blob/main/images/SDBox-v2_VBCC_pic4.jpg

**Comment by Alectronics** [comment]
Yes of course I've already seen this but there is no help in how to start the on my desktop installed vbcc map? There are no icons either?
________________________________
Van: Jorgen Bilander ***@***.***>
Verzonden: zondag 6 augustus 2023 12:50
Aan: jbilander/SDBox-v2 ***@***.***>
CC: Alectronics ***@***.***>; Author ***@***.***>
Onderwerp: Re: [jbilander/SDBox-v2] Build mini version 2B with smd components and Atmega328p chip. But all sd cards are No dos recognised! (Issue #7)


All instructions are there, nothing is hidden, did you even look??? did you even try??? did you even check activity on the signals I told you to check??? Sorry to say but a DIY-project is not a thing for you if you need to be spoon-fed.

[Screen Shot 2023-08-06 at 12 37 00]<https://user-images.githubusercontent.com/1673918/258636728-e389c68f-04e8-42fb-a54b-86b1ca1df6e4.png>

https://github.com/jbilander/SDBox-v2/blob/main/images/SDBox-v2_VBCC_pic1.jpg

https://github.com/jbilander/SDBox-v2/blob/main/images/SDBox-v2_VBCC_pic2.jpg

https://github.com/jbilander/SDBox-v2/blob/main/images/SDBox-v2_VBCC_pic3.jpg

https://github.com/jbilander/SDBox-v2/blob/main/images/SDBox-v2_VBCC_pic4.jpg

**Comment by Alectronics** [comment]
Writing the new Marlin firmware for my Ender 5 Pro 3D printer last year was very easy with Microsoft's Visual Studio because all files and clear instructions on how to do that made it plain and simple. And my 3D printer still works better as new after all hardware added upgrades. And I'm sorry but here nothing is made that clear and easy. Why so many problems exist with this SD0: project and so many SD cards fail including the miniSDBox cardreader itself.

**Comment by Alectronics** [comment]
Hi Jörgen,
Although I still have no clue how to use any of the files in VBCC I managed to use your Forked main.hex file that makes use of the red led. (by simply running main.bat in Windows first).
And flashed main.hex with my TL866II into my mini SDbox board.
Further sadly still no progress at all.
The mini sd card reader writer still keeps saying that there is no dos disk in the reader every time I extract and put back in the sd card. So I still have no useful working reader but I probably can be sure that my board IS working fine.
And the sd0: icon is also nowhere visible on the desktop either yet.

Isn't there anyone who wrote about how to use the VBCC with the Amiga SD card files in a way that it makes some sense and works?
I know how to start vc.exe in the cmd popup screen but have no clue what to do with it next.

Albert.

________________________________
Van: Jorgen Bilander ***@***.***>
Verzonden: zondag 6 augustus 2023 12:50
Aan: jbilander/SDBox-v2 ***@***.***>
CC: Alectronics ***@***.***>; Author ***@***.***>
Onderwerp: Re: [jbilander/SDBox-v2] Build mini version 2B with smd components and Atmega328p chip. But all sd cards are No dos recognised! (Issue #7)


All instructions are there, nothing is hidden, did you even look??? did you even try??? did you even check activity on the signals I told you to check??? Sorry to say but a DIY-project is not a thing for you if you need to be spoon-fed.

[Screen Shot 2023-08-06 at 12 37 00]<https://user-images.githubusercontent.com/1673918/258636728-e389c68f-04e8-42fb-a54b-86b1ca1df6e4.png>

https://github.com/jbilander/SDBox-v2/blob/main/images/SDBox-v2_VBCC_pic1.jpg

https://github.com/jbilander/SDBox-v2/blob/main/images/SDBox-v2_VBCC_pic2.jpg

https://github.com/jbilander/SDBox-v2/blob/main/images/SDBox-v2_VBCC_pic3.jpg

https://github.com/jbilander/SDBox-v2/blob/main/images/SDBox-v2_VBCC_pic4.jpg

**Comment by Alectronics** [comment]
PS: maybe the sd card pullup and pull downs being here 10K's are wrong. Because this Pololu micro SD arduino compatile breakout board uses ALL 100K's for pullup and pulldown resistors! See: https://www.pololu.com/product/2587

**Comment by jbilander** [comment]
It's 10-100kOhm for the CMD line and DAT lines. If DAT3 is used for card detection, that pull-up needs to be 10-90kOhm. CLK does not need a pullup. 

The extended DAT lines (DAT1-DAT3) are input on power up. They start to operate as DAT lines after the SET_BUS_WIDTH command. It is the responsibility of the host designer to connect external pullup resistors to all data lines even if only DAT0 is to be used. Otherwise, non-expected high current consumption may occur due to the floating inputs of DAT1 & DAT2 (in case they are not used).

<img width="895" alt="Screen Shot 2023-08-11 at 11 47 31" src="https://github.com/jbilander/SDBox-v2/assets/1673918/a94123d0-f011-4d21-add7-6fe43339f0a6">

**Comment by Alectronics** [comment]
Yes, but did you also noticed the 100K pulldown resistor on the CLK input schmitt-trigger? It isn't there in the miniBox circuit.

**Comment by jbilander** [comment]
It's far fetched that pull-down makes any difference. According to the spec CLK should not have any pull-up or down resistors. It's more likely you'll need pull-ups on the parallel-lines, like this:

https://github.com/jbilander/sdbox/issues/18#issuecomment-1549666472

and of course you'll need the correct _spisd.device_ in Devs: and the correct version of SD0-file for your WB-version together with the latest fat95 binary in L:, and the firmware flashed successfully to the AVR running at 16 MHz.

**Comment by Alectronics** [comment]
If Fat95 v3.18 was the latest I'm sure I did! I also used a special downloaded v2.2 SDbox boot disk that worked fine after clicking on the mount_Sdcard icon. But only resulted again in a No Dos message. And it also said SD0: already loaded but it never shows its icon. Only the Nodos disk message in a small popup window. Also the red activity led flashes twice when re-inserting the microSD card. So your forked main.hex was apparently flashed okay.

**Comment by Alectronics** [comment]
The SD extension library is entirely added into the Arduino IDE software and there were never reported such extreme compatibility issues. The Arduino IDE is likely correctly working with reading and writing to ALL original PC FAT/FAT32 SD cards without any read/write errors whatsoever. The existing SDbox SD card problem is therefore unacceptable! Really tried every card including 64GB Sandisk formatted as FAT16 and FAT32 at 512MB etc. getting only NO DOS errors. This is by no means workable fun ready firmware! Never worked! And leaving out any crc/parity data writing/reading checks is a bad idea too! If it even did work!

**Comment by jbilander** [comment]
Another SDBox-Mini built, works like a charm...

![IMG_8419](https://github.com/jbilander/SDBox-v2/assets/1673918/087ee10a-ea92-43d8-b867-7fc3ceb758ae)

built and flashed the fw as before by running _build.bat_ and then _flashSDBoxMini.bat_, no problem what so ever:

Video below, +5V taken from floppy port:
https://drive.google.com/file/d/1fLwmIyUeHZX4sln5bpJfz5QkDGD-QTOf/view?usp=sharing


    C:\Users\Jorgen\Projects\FORKED_amiga-par-to-spi-adapter\avr>build.bat

    C:\Users\Jorgen\Projects\FORKED_amiga-par-to-spi-adapter\avr>"C:\\Program Files (x86)\\Arduino\\hardware\\tools\\avr/bin/avr-gcc" -Os -mmcu=atmega328p main.c -o main.elf

    C:\Users\Jorgen\Projects\FORKED_amiga-par-to-spi-adapter\avr>"C:\\Program Files (x86)\\Arduino\\hardware\\tools\\avr/bin/avr-objcopy" -O ihex main.elf main.hex


    C:\Users\Jorgen\Projects\FORKED_amiga-par-to-spi-adapter\avr>flashSDBoxMini.bat

    C:\Users\Jorgen\Projects\FORKED_amiga-par-to-spi-adapter\avr>"C:\Program Files (x86)\Arduino\hardware\tools\avr/bin/avrdude" -C"C:\Program Files (x86)\Arduino\hardware\tools\avr/et
    c/avrdude.conf" -v -pm328p -cstk500v1 -PCOM3 -b19200 -Uflash:w:main.hex:i -Ulfuse:w:0xe6:m -Uhfuse:w:0xdf:m -Uefuse:w:0xfc:m

    avrdude: Version 6.3-20190619
             Copyright (c) 2000-2005 Brian Dean, http://www.bdmicro.com/
             Copyright (c) 2007-2014 Joerg Wunsch

             System wide configuration file is "C:\Program Files (x86)\Arduino\hardware\tools\avr/etc/avrdude.conf"

             Using Port                    : COM3
             Using Programmer              : stk500v1
             Overriding Baud Rate          : 19200
             AVR Part                      : ATmega328P
             Chip Erase delay              : 9000 us
             PAGEL                         : PD7
             BS2                           : PC2
             RESET disposition             : dedicated
             RETRY pulse                   : SCK
             serial program mode           : yes
             parallel program mode         : yes
             Timeout                       : 200
             StabDelay                     : 100
             CmdexeDelay                   : 25
             SyncLoops                     : 32
             ByteDelay                     : 0
             PollIndex                     : 3
             PollValue                     : 0x53
             Memory Detail                 :

                                      Block Poll               Page                       Polled
               Memory Type Mode Delay Size  Indx Paged  Size   Size #Pages MinW  MaxW   ReadBack
               ----------- ---- ----- ----- ---- ------ ------ ---- ------ ----- ----- ---------
               eeprom        65    20     4    0 no       1024    4      0  3600  3600 0xff 0xff
               flash         65     6   128    0 yes     32768  128    256  4500  4500 0xff 0xff
               lfuse          0     0     0    0 no          1    0      0  4500  4500 0x00 0x00
               hfuse          0     0     0    0 no          1    0      0  4500  4500 0x00 0x00
               efuse          0     0     0    0 no          1    0      0  4500  4500 0x00 0x00
               lock           0     0     0    0 no          1    0      0  4500  4500 0x00 0x00
               calibration    0     0     0    0 no          1    0      0     0     0 0x00 0x00
               signature      0     0     0    0 no          3    0      0     0     0 0x00 0x00

             Programmer Type : STK500
             Description     : Atmel STK500 Version 1.x firmware
             Hardware Version: 2
             Firmware Version: 1.18
             Topcard         : Unknown
             Vtarget         : 0.0 V
             Varef           : 0.0 V
             Oscillator      : Off
             SCK period      : 0.1 us

    avrdude: AVR device initialized and ready to accept instructions

    Reading | ################################################## | 100% 0.08s

    avrdude: Device signature = 0x1e950f (probably m328p)
    avrdude: safemode: lfuse reads as E6
    avrdude: safemode: hfuse reads as DF
    avrdude: safemode: efuse reads as FC
    avrdude: NOTE: "flash" memory has been specified, an erase cycle will be performed
             To disable this feature, specify the -D option.
    avrdude: erasing chip
    avrdude: reading input file "main.hex"
    avrdude: writing flash (496 bytes):

    Writing | ################################################## | 100% 0.69s

    avrdude: 496 bytes of flash written
    avrdude: verifying flash memory against main.hex:
    avrdude: load data flash data from input file main.hex:
    avrdude: input file main.hex contains 496 bytes
    avrdude: reading on-chip flash data:

    Reading | ################################################## | 100% 0.41s

    avrdude: verifying ...
    avrdude: 496 bytes of flash verified
    avrdude: reading input file "0xe6"
    avrdude: writing lfuse (1 bytes):

    Writing | ################################################## | 100% 0.02s

    avrdude: 1 bytes of lfuse written
    avrdude: verifying lfuse memory against 0xe6:
    avrdude: load data lfuse data from input file 0xe6:
    avrdude: input file 0xe6 contains 1 bytes
    avrdude: reading on-chip lfuse data:

    Reading | ################################################## | 100% 0.02s

    avrdude: verifying ...
    avrdude: 1 bytes of lfuse verified
    avrdude: reading input file "0xdf"
    avrdude: writing hfuse (1 bytes):

    Writing | ################################################## | 100% 0.02s

    avrdude: 1 bytes of hfuse written
    avrdude: verifying hfuse memory against 0xdf:
    avrdude: load data hfuse data from input file 0xdf:
    avrdude: input file 0xdf contains 1 bytes
    avrdude: reading on-chip hfuse data:

    Reading | ################################################## | 100% 0.02s

    avrdude: verifying ...
    avrdude: 1 bytes of hfuse verified
    avrdude: reading input file "0xfc"
    avrdude: writing efuse (1 bytes):

    Writing | ################################################## | 100% 0.02s

    avrdude: 1 bytes of efuse written
    avrdude: verifying efuse memory against 0xfc:
    avrdude: load data efuse data from input file 0xfc:
    avrdude: input file 0xfc contains 1 bytes
    avrdude: reading on-chip efuse data:

    Reading | ################################################## | 100% 0.02s

    avrdude: verifying ...
    avrdude: 1 bytes of efuse verified

    avrdude: safemode: lfuse reads as E6
    avrdude: safemode: hfuse reads as DF
    avrdude: safemode: efuse reads as FC
    avrdude: safemode: Fuses OK (E:FC, H:DF, L:E6)

    avrdude done.  Thank you.

    C:\Users\Jorgen\Projects\FORKED_amiga-par-to-spi-adapter\avr>

**Comment by Alectronics** [comment]
I used my TL866-II with your main.hex Jorgen. That may be an older version why it failed working?
Do I need to use that FlashSDmini.bat file to get it working too?
Because I didn't use my Arduino uno or whatever board to copy flash the file over into my minSDbox 328p processor. And maybe that is the only difference here?
I however used the correct configuration fuses for the given main.hex and it verified fine! Also your forked red led routine worked afterwards. Only reading any SD card gave still only No Dos error messages.

So I am now waiting on my version 1 PCBway boards hoping that then that SDbox reader will work.

Best regards.




________________________________
Van: Jorgen Bilander ***@***.***>
Verzonden: dinsdag 15 augustus 2023 16:13
Aan: jbilander/SDBox-v2 ***@***.***>
CC: Alectronics ***@***.***>; Author ***@***.***>
Onderwerp: Re: [jbilander/SDBox-v2] Build mini version 2B with smd components and Atmega328p chip. But all sd cards are No dos recognised! (Issue #7)


Another SDBox-Mini built, works like a charm...

[IMG_8419]<https://user-images.githubusercontent.com/1673918/260732133-087ee10a-ea92-43d8-b867-7fc3ceb758ae.JPG>

built and flashed the fw as before by running build.bat and then flashSDBoxMini.bat, no problem what so ever:

Video below, +5V taken from floppy port:
https://drive.google.com/file/d/1fLwmIyUeHZX4sln5bpJfz5QkDGD-QTOf/view?usp=sharing

C:\Users\Jorgen\Projects\FORKED_amiga-par-to-spi-adapter\avr>build.bat

C:\Users\Jorgen\Projects\FORKED_amiga-par-to-spi-adapter\avr>"C:\\Program Files (x86)\\Arduino\\hardware\\tools\\avr/bin/avr-gcc" -Os -mmcu=atmega328p main.c -o main.elf

C:\Users\Jorgen\Projects\FORKED_amiga-par-to-spi-adapter\avr>"C:\\Program Files (x86)\\Arduino\\hardware\\tools\\avr/bin/avr-objcopy" -O ihex main.elf main.hex


C:\Users\Jorgen\Projects\FORKED_amiga-par-to-spi-adapter\avr>flashSDBoxMini.bat

C:\Users\Jorgen\Projects\FORKED_amiga-par-to-spi-adapter\avr>"C:\Program Files (x86)\Arduino\hardware\tools\avr/bin/avrdude" -C"C:\Program Files (x86)\Arduino\hardware\tools\avr/et
c/avrdude.conf" -v -pm328p -cstk500v1 -PCOM3 -b19200 -Uflash:w:main.hex:i -Ulfuse:w:0xe6:m -Uhfuse:w:0xdf:m -Uefuse:w:0xfc:m

avrdude: Version 6.3-20190619
         Copyright (c) 2000-2005 Brian Dean, http://www.bdmicro.com/
         Copyright (c) 2007-2014 Joerg Wunsch

         System wide configuration file is "C:\Program Files (x86)\Arduino\hardware\tools\avr/etc/avrdude.conf"

         Using Port                    : COM3
         Using Programmer              : stk500v1
         Overriding Baud Rate          : 19200
         AVR Part                      : ATmega328P
         Chip Erase delay              : 9000 us
         PAGEL                         : PD7
         BS2                           : PC2
         RESET disposition             : dedicated
         RETRY pulse                   : SCK
         serial program mode           : yes
         parallel program mode         : yes
         Timeout                       : 200
         StabDelay                     : 100
         CmdexeDelay                   : 25
         SyncLoops                     : 32
         ByteDelay                     : 0
         PollIndex                     : 3
         PollValue                     : 0x53
         Memory Detail                 :

                                  Block Poll               Page                       Polled
           Memory Type Mode Delay Size  Indx Paged  Size   Size #Pages MinW  MaxW   ReadBack
           ----------- ---- ----- ----- ---- ------ ------ ---- ------ ----- ----- ---------
           eeprom        65    20     4    0 no       1024    4      0  3600  3600 0xff 0xff
           flash         65     6   128    0 yes     32768  128    256  4500  4500 0xff 0xff
           lfuse          0     0     0    0 no          1    0      0  4500  4500 0x00 0x00
           hfuse          0     0     0    0 no          1    0      0  4500  4500 0x00 0x00
           efuse          0     0     0    0 no          1    0      0  4500  4500 0x00 0x00
           lock           0     0     0    0 no          1    0      0  4500  4500 0x00 0x00
           calibration    0     0     0    0 no          1    0      0     0     0 0x00 0x00
           signature      0     0     0    0 no          3    0      0     0     0 0x00 0x00

         Programmer Type : STK500
         Description     : Atmel STK500 Version 1.x firmware
         Hardware Version: 2
         Firmware Version: 1.18
         Topcard         : Unknown
         Vtarget         : 0.0 V
         Varef           : 0.0 V
         Oscillator      : Off
         SCK period      : 0.1 us

avrdude: AVR device initialized and ready to accept instructions

Reading | ################################################## | 100% 0.08s

avrdude: Device signature = 0x1e950f (probably m328p)
avrdude: safemode: lfuse reads as E6
avrdude: safemode: hfuse reads as DF
avrdude: safemode: efuse reads as FC
avrdude: NOTE: "flash" memory has been specified, an erase cycle will be performed
         To disable this feature, specify the -D option.
avrdude: erasing chip
avrdude: reading input file "main.hex"
avrdude: writing flash (496 bytes):

Writing | ################################################## | 100% 0.69s

avrdude: 496 bytes of flash written
avrdude: verifying flash memory against main.hex:
avrdude: load data flash data from input file main.hex:
avrdude: input file main.hex contains 496 bytes
avrdude: reading on-chip flash data:

Reading | ################################################## | 100% 0.41s

avrdude: verifying ...
avrdude: 496 bytes of flash verified
avrdude: reading input file "0xe6"
avrdude: writing lfuse (1 bytes):

Writing | ################################################## | 100% 0.02s

avrdude: 1 bytes of lfuse written
avrdude: verifying lfuse memory against 0xe6:
avrdude: load data lfuse data from input file 0xe6:
avrdude: input file 0xe6 contains 1 bytes
avrdude: reading on-chip lfuse data:

Reading | ################################################## | 100% 0.02s

avrdude: verifying ...
avrdude: 1 bytes of lfuse verified
avrdude: reading input file "0xdf"
avrdude: writing hfuse (1 bytes):

Writing | ################################################## | 100% 0.02s

avrdude: 1 bytes of hfuse written
avrdude: verifying hfuse memory against 0xdf:
avrdude: load data hfuse data from input file 0xdf:
avrdude: input file 0xdf contains 1 bytes
avrdude: reading on-chip hfuse data:

Reading | ################################################## | 100% 0.02s

avrdude: verifying ...
avrdude: 1 bytes of hfuse verified
avrdude: reading input file "0xfc"
avrdude: writing efuse (1 bytes):

Writing | ################################################## | 100% 0.02s

avrdude: 1 bytes of efuse written
avrdude: verifying efuse memory against 0xfc:
avrdude: load data efuse data from input file 0xfc:
avrdude: input file 0xfc contains 1 bytes
avrdude: reading on-chip efuse data:

Reading | ################################################## | 100% 0.02s

avrdude: verifying ...
avrdude: 1 bytes of efuse verified

avrdude: safemode: lfuse reads as E6
avrdude: safemode: hfuse reads as DF
avrdude: safemode: efuse reads as FC
avrdude: safemode: Fuses OK (E:FC, H:DF, L:E6)

avrdude done.  Thank you.

C:\Users\Jorgen\Projects\FORKED_amiga-par-to-spi-adapter\avr>

**Comment by Alectronics** [comment]
And I always use the nicer AVRdudess so do not know if AVRdude works for me. Never successfully used AVRdude before.
(I always flash my firmware upgrades into my RCT chiptester with a Pololu programmer v2.1 into an Arduino ATmega 2560 processor).

________________________________
Van: Albert van bemmelen ***@***.***>
Verzonden: dinsdag 15 augustus 2023 16:41
Aan: jbilander/SDBox-v2 ***@***.***>
Onderwerp: Re: [jbilander/SDBox-v2] Build mini version 2B with smd components and Atmega328p chip. But all sd cards are No dos recognised! (Issue #7)

I used my TL866-II with your main.hex Jorgen. That may be an older version why it failed working?
Do I need to use that FlashSDmini.bat file to get it working too?
Because I didn't use my Arduino uno or whatever board to copy flash the file over into my minSDbox 328p processor. And maybe that is the only difference here?
I however used the correct configuration fuses for the given main.hex and it verified fine! Also your forked red led routine worked afterwards. Only reading any SD card gave still only No Dos error messages.

So I am now waiting on my version 1 PCBway boards hoping that then that SDbox reader will work.

Best regards.




________________________________
Van: Jorgen Bilander ***@***.***>
Verzonden: dinsdag 15 augustus 2023 16:13
Aan: jbilander/SDBox-v2 ***@***.***>
CC: Alectronics ***@***.***>; Author ***@***.***>
Onderwerp: Re: [jbilander/SDBox-v2] Build mini version 2B with smd components and Atmega328p chip. But all sd cards are No dos recognised! (Issue #7)


Another SDBox-Mini built, works like a charm...

[IMG_8419]<https://user-images.githubusercontent.com/1673918/260732133-087ee10a-ea92-43d8-b867-7fc3ceb758ae.JPG>

built and flashed the fw as before by running build.bat and then flashSDBoxMini.bat, no problem what so ever:

Video below, +5V taken from floppy port:
https://drive.google.com/file/d/1fLwmIyUeHZX4sln5bpJfz5QkDGD-QTOf/view?usp=sharing

C:\Users\Jorgen\Projects\FORKED_amiga-par-to-spi-adapter\avr>build.bat

C:\Users\Jorgen\Projects\FORKED_amiga-par-to-spi-adapter\avr>"C:\\Program Files (x86)\\Arduino\\hardware\\tools\\avr/bin/avr-gcc" -Os -mmcu=atmega328p main.c -o main.elf

C:\Users\Jorgen\Projects\FORKED_amiga-par-to-spi-adapter\avr>"C:\\Program Files (x86)\\Arduino\\hardware\\tools\\avr/bin/avr-objcopy" -O ihex main.elf main.hex


C:\Users\Jorgen\Projects\FORKED_amiga-par-to-spi-adapter\avr>flashSDBoxMini.bat

C:\Users\Jorgen\Projects\FORKED_amiga-par-to-spi-adapter\avr>"C:\Program Files (x86)\Arduino\hardware\tools\avr/bin/avrdude" -C"C:\Program Files (x86)\Arduino\hardware\tools\avr/et
c/avrdude.conf" -v -pm328p -cstk500v1 -PCOM3 -b19200 -Uflash:w:main.hex:i -Ulfuse:w:0xe6:m -Uhfuse:w:0xdf:m -Uefuse:w:0xfc:m

avrdude: Version 6.3-20190619
         Copyright (c) 2000-2005 Brian Dean, http://www.bdmicro.com/
         Copyright (c) 2007-2014 Joerg Wunsch

         System wide configuration file is "C:\Program Files (x86)\Arduino\hardware\tools\avr/etc/avrdude.conf"

         Using Port                    : COM3
         Using Programmer              : stk500v1
         Overriding Baud Rate          : 19200
         AVR Part                      : ATmega328P
         Chip Erase delay              : 9000 us
         PAGEL                         : PD7
         BS2                           : PC2
         RESET disposition             : dedicated
         RETRY pulse                   : SCK
         serial program mode           : yes
         parallel program mode         : yes
         Timeout                       : 200
         StabDelay                     : 100
         CmdexeDelay                   : 25
         SyncLoops                     : 32
         ByteDelay                     : 0
         PollIndex                     : 3
         PollValue                     : 0x53
         Memory Detail                 :

                                  Block Poll               Page                       Polled
           Memory Type Mode Delay Size  Indx Paged  Size   Size #Pages MinW  MaxW   ReadBack
           ----------- ---- ----- ----- ---- ------ ------ ---- ------ ----- ----- ---------
           eeprom        65    20     4    0 no       1024    4      0  3600  3600 0xff 0xff
           flash         65     6   128    0 yes     32768  128    256  4500  4500 0xff 0xff
           lfuse          0     0     0    0 no          1    0      0  4500  4500 0x00 0x00
           hfuse          0     0     0    0 no          1    0      0  4500  4500 0x00 0x00
           efuse          0     0     0    0 no          1    0      0  4500  4500 0x00 0x00
           lock           0     0     0    0 no          1    0      0  4500  4500 0x00 0x00
           calibration    0     0     0    0 no          1    0      0     0     0 0x00 0x00
           signature      0     0     0    0 no          3    0      0     0     0 0x00 0x00

         Programmer Type : STK500
         Description     : Atmel STK500 Version 1.x firmware
         Hardware Version: 2
         Firmware Version: 1.18
         Topcard         : Unknown
         Vtarget         : 0.0 V
         Varef           : 0.0 V
         Oscillator      : Off
         SCK period      : 0.1 us

avrdude: AVR device initialized and ready to accept instructions

Reading | ################################################## | 100% 0.08s

avrdude: Device signature = 0x1e950f (probably m328p)
avrdude: safemode: lfuse reads as E6
avrdude: safemode: hfuse reads as DF
avrdude: safemode: efuse reads as FC
avrdude: NOTE: "flash" memory has been specified, an erase cycle will be performed
         To disable this feature, specify the -D option.
avrdude: erasing chip
avrdude: reading input file "main.hex"
avrdude: writing flash (496 bytes):

Writing | ################################################## | 100% 0.69s

avrdude: 496 bytes of flash written
avrdude: verifying flash memory against main.hex:
avrdude: load data flash data from input file main.hex:
avrdude: input file main.hex contains 496 bytes
avrdude: reading on-chip flash data:

Reading | ################################################## | 100% 0.41s

avrdude: verifying ...
avrdude: 496 bytes of flash verified
avrdude: reading input file "0xe6"
avrdude: writing lfuse (1 bytes):

Writing | ################################################## | 100% 0.02s

avrdude: 1 bytes of lfuse written
avrdude: verifying lfuse memory against 0xe6:
avrdude: load data lfuse data from input file 0xe6:
avrdude: input file 0xe6 contains 1 bytes
avrdude: reading on-chip lfuse data:

Reading | ################################################## | 100% 0.02s

avrdude: verifying ...
avrdude: 1 bytes of lfuse verified
avrdude: reading input file "0xdf"
avrdude: writing hfuse (1 bytes):

Writing | ################################################## | 100% 0.02s

avrdude: 1 bytes of hfuse written
avrdude: verifying hfuse memory against 0xdf:
avrdude: load data hfuse data from input file 0xdf:
avrdude: input file 0xdf contains 1 bytes
avrdude: reading on-chip hfuse data:

Reading | ################################################## | 100% 0.02s

avrdude: verifying ...
avrdude: 1 bytes of hfuse verified
avrdude: reading input file "0xfc"
avrdude: writing efuse (1 bytes):

Writing | ################################################## | 100% 0.02s

avrdude: 1 bytes of efuse written
avrdude: verifying efuse memory against 0xfc:
avrdude: load data efuse data from input file 0xfc:
avrdude: input file 0xfc contains 1 bytes
avrdude: reading on-chip efuse data:

Reading | ################################################## | 100% 0.02s

avrdude: verifying ...
avrdude: 1 bytes of efuse verified

avrdude: safemode: lfuse reads as E6
avrdude: safemode: hfuse reads as DF
avrdude: safemode: efuse reads as FC
avrdude: safemode: Fuses OK (E:FC, H:DF, L:E6)

avrdude done.  Thank you.

C:\Users\Jorgen\Projects\FORKED_amiga-par-to-spi-adapter\avr>

**Comment by Alectronics** [comment]
I don't know why but I think my miniSDbox board is just defect because now when I tried to program my reader on the 6 pin Arduino ISP connector with my perfectly working STK500 v1 compatible Pololu v2.1 I just can't! Because as soon as I connect the 6 pins to my reader the Pololu gets blocked in its hardware and its configuration window. So my processor or some other part may be defect. The yellow programming led no longer works when the ISP programming voltage is connected like it did before and it only works when the external USB cable voltage is 5V fed. So my PCBway board or one of the chips may be defect why I no longer am able to reprogram it. My TL866-II probably still can because it did before successfully. But no idea why my Pololu fails on ISP programming the processor now. I never did anything wrong and also my CIA chip on my Amiga Parallelport is still fine. So I stop building & testing this failing miniSDbox project. It never worked and likely never will. I hope my version 1 boards from PCBway are stronger and are made with a much better successrate!

**Comment by Alectronics** [comment]
Hi Jorgen, today I used the main.c code to flash the Nano 3.0 board for a version 1A SDbox  in Arduino IDE.
But that didn't seem to work on my Amiga1200 yet?

I first had to find out why my 4 new Nano 3.0 boards from China were not communicating in the Arduino IDE. Not over USB and not with ISP. Brandnew they only blinked and were completely useless.

I solved the issue in below explanation but still couldn't use your main.hex and flash.bat program with my Pololu as STK500 v2 afterwards because I only got 10 times a STK500 not connecting error.


My 4 Nano 3.0 CH340 ATmega328p boards were completely dead not communicating at all!
But after 2 days of dismay I found how to solve the 4 previous useless Nano 3.0 boards issue.

luckily I found and solved the problem today. With my TL866II I flashed the new optiboot Arduino 328p bootfile from ARDUINO IDE over into the bad not communicating only flashing L led and lit PWR led NANO 3.0 defect looking boards. ISP connected. That previously were not communicating over USB and neither on ISP!
BUT WHILE AND DURING THE PROGRAMMING the button on the NANO 3.0 module must be pressed in all the time! Else the Nano 3.0 boards are completely dead and are wrong read as being some strange $FFFFFFFF processor! Now the modules are working and correctly read as 0x1E950F processor boards! And can without any problem be used in any ARDUINO IDE program.



I hope you can tell me if the MAIN.c SDbox v1a file copied into Arduino IDE not working is to be expected for some reason?

Please tell me Jorgen if there is another way to copy the MAIN.hex file into the now working Nano 3.0 board that will not fail and work?

Best regards.

Albert.

**Comment by jbilander** [comment]
On the Arduino Nano V3 which is used in SDBox-v1 and v2 I just flash the AVR via connecting usb and running flash.bat. You may need to change to correct com-port and baud speed in the bat-file.

For SDBox-v1 you need the Main.hex from here, look in the avr folder after unzip.
https://github.com/jbilander/sdbox/releases

**Comment by Alectronics** [comment]
Thanks Jorgen that worked superfast. Is het correct only 488 bytes? Please see my screencopy. Nano 3 with the main.hex file flashed.

I hope SD Box 1a now works.

Albert.

________________________________
Van: Jorgen Bilander ***@***.***>
Verzonden: maandag 21 augustus 2023 20:16
Aan: jbilander/SDBox-v2 ***@***.***>
CC: Alectronics ***@***.***>; Author ***@***.***>
Onderwerp: Re: [jbilander/SDBox-v2] Build mini version 2B with smd components and Atmega328p chip. But all sd cards are No dos recognised! (Issue #7)


On the Arduino Nano V3 which is used in SDBox-v1 and v2 I just flash the AVR via connecting usb and running flash.bat. You may need to change to correct com-port and baud speed in the bat-file.

For SDBox-v1 you need the Main.hex from here, look in the avr folder after unzip.
https://github.com/jbilander/sdbox/releases

**Comment by jbilander** [comment]
Yes that sounds correct. You can compare with the documentation. It was 410 bytes only at one time but that screenshot is a bit old...

https://github.com/jbilander/sdbox/tree/master/avr

**Comment by Alectronics** [comment]
I happy that also worked! But am now still waiting on the tiny smd transistors before I will be able to test the SDbox 1a board with both leds.

And I suppose that my in the miniSDbox 2.2b good working SD Fat32 2GB cards are a problem in the version 1 SD box because it now didn't work yet when tested with them on my Amiga1200. Because SD0: didn't show up yet. But I suppose SDBox 1a finally will work when everything is soldered in.

Anyway many thanks for the great help you provided Jorgen!

Albert.

________________________________
Van: Jorgen Bilander ***@***.***>
Verzonden: maandag 21 augustus 2023 20:41
Aan: jbilander/SDBox-v2 ***@***.***>
CC: Alectronics ***@***.***>; Author ***@***.***>
Onderwerp: Re: [jbilander/SDBox-v2] Build mini version 2B with smd components and Atmega328p chip. But all sd cards are No dos recognised! (Issue #7)


Yes that sounds correct. You can compare with the documentation. It was 410 bytes only at one time but that screenshot is a bit old...

https://github.com/jbilander/sdbox/tree/master/avr

**Comment by jbilander** [comment]
It should work without the transistors soldered, they are optional, but you need to use the correct _spisd.device_ driver for v1 on the Amiga side, they v1 and v2 are not compatible with each other hence why we bumped main version number.

**Comment by Alectronics** [comment]
Understood Jorgen. But I now used two different bootdisks for SDbox v1 to eliminate that doubt to correctly test the device.

Albert.

________________________________
Van: Jorgen Bilander ***@***.***>
Verzonden: maandag 21 augustus 2023 21:17
Aan: jbilander/SDBox-v2 ***@***.***>
CC: Alectronics ***@***.***>; Author ***@***.***>
Onderwerp: Re: [jbilander/SDBox-v2] Build mini version 2B with smd components and Atmega328p chip. But all sd cards are No dos recognised! (Issue #7)


It should work without the transistors soldered, they are optional, but you need to use the correct spisd.device driver for v1 on the Amiga side, they v1 and v2 are not compatible with each other hence why we bumped main version number.

**Comment by Alectronics** [comment]
Hi Jorgen,

Just to let you know, my older SDbox version tested working good today too with a 512MB FAT partitioned 2GB Sandisk microSD card.
Which I made too as you know only because at first my miniSDbox v2.2B gave a NoDos error message because of the defect 5 pin smd buffer IC.
That I fixed by replacing the bad by my TL866II blown buffer chip while successfully flashing the ATmega 328p processor.

Thanks!
Albert.
